Feature/prettier

miniduikboot opened the pull request:
This commit introduces Prettier to the codebase and reformats it. It also adds multiple safeguards to keep styling correct in future: precommit hooks will make sure new code is formatted, while CI will make sure the entire repository remains formatted.

Tabs were chosen to indent with to make it possible for everyone to use their own preferred tab size. Previously the code base was a mix of 2 and 4 spaces.

Commit 1 adds prettier to the repository. It also removed the formatting rules from webui. A configuration package for eslint is also added that disables all styling related rules from eslint in favor of letting prettier run them.

Commit 2 can be reproduced by running `yarn prettier --write .` on a local checkout of commit 1. I should maybe change the author to prevent this to get attributed to me.

Commit 3 adds a precommit hook using husky and lint-staged, which was the [first recommended option from Prettier's docs](https://prettier.io/docs/en/precommit.html). I changed the hook to call `yarn` instead of `npx`.

Commit 4 runs eslint and prettier in CI. Currently these steps are fatal and should fail the CI completely. We should monitor how this works and if they should become optional in future.

miniduikboot committed:
Enable prettier on the entire repo

This commit only makes the necessary changes to the rules and can be
manually reviewed

miniduikboot committed:
Format the entire repo with Prettier

miniduikboot committed:
Add Husky pre-commit hook

miniduikboot committed:
Enforce code style in Github Actions

edqx commented:
I think it's better for the linting & building to be on 2 separate workflows since they're not reliant on each other being successful.

Also check the run & I'll merge.

miniduikboot committed:
Symlink .gitignore to .prettierignore

Files we don't track in git don't need to be formatted. Notably this
includes the `dist/` folder with compiled artifacts. Symlink because I
don't see why these files should be different.

miniduikboot committed:
Split linting and building into separate workflows

edqx commented:
No cryptominers here.
